Unity-Technologies/Graphics
# Give the inspector view a "show shader code" button.

## Purpose of this PR
This change adds a "View Generated Shader" button to the editor inspector when a shader graph asset is selected.

Viewing the generated shader:
- writes a temp file with the generated shader code
- opens the temp file in the code editing tool

![2022-03-02 10 04 53](https://user-images.githubusercontent.com/96441566/156422016-f124f794-5a0f-44d4-bb41-1f40fdc45ceb.gif)

---
## Testing status
UI element; Manual testing only.

### Steps to Test
1. In the Project view, right-click and "Create" > "Shader Graph" > "Blank Shader Graph"
2. Select the shader graph that you created in the Project View
  Notice: The inspector updates and has a "View Generated Shader" button.
3. In the inspector, left-click on "View Generated Shader"
  Notice: The generated shader code appears in a code editor window.

---
## Comments to reviewers
This PR does not port the other buttons that are available in the Current inspector view for a shader graph asset. Those will be ported as needed.
